### ISSUE TITLE
[Ubuntu 18.04] Segfault while running qemu-system-cheri128

### ISSUE THREAD

**Issue by bharadwajy** [CTSRD-CHERI/cheribuild#96]
Hello!

I'm see a segmentation fault upon executing the following command:

```
$ ./cheribuild.py run-mips-purecap -d --config-file /home/bharadwaj/datadrive/github/src/CHERI/.config/cheribuild.json --source-root /home/bharadwaj/datadrive/github/src/CHERI

...

Listening for SSH connections on localhost:10005
...
Fatal error: Command `/home/bharadwaj/datadrive/github/src/CHERI/output/sdk/bin/qemu-system-cheri128 -M malta -kernel /home/bharadwaj/datadrive/github/src/CHERI/output/rootfs-purecap128/boot/kernel/kernel -m 2048 -nographic -drive file=/home/bharadwaj/datadrive/github/src/CHERI/output/cheribsd-mips-purecap128.img,format=raw,index=0,media=disk -net nic -net 'user,id=net0,ipv6=off,smb=/home/bharadwaj/datadrive/github/src/CHERI<<<source_root@ro:/home/bharadwaj/datadrive/github/src/CHERI/build<<<build_root:/home/bharadwaj/datadrive/github/src/CHERI/output<<<output_root@ro:/home/bharadwaj/datadrive/github/src/CHERI/output/rootfs-purecap128<<<rootfs,hostfwd=tcp::10005-:22' -device virtio-rng-pci` failed with non-zero exit code -11. Working directory was /mnt/datadrive/github/src/CHERI/cheribuild

```

Tip of cheribuild repo used:
```
commit 2a799388e87562913801cba33f5199d251bc596c (HEAD -> master, origin/master, origin/HEAD)
Author: Alex Richardson <Alexander.Richardson@cl.cam.ac.uk>
Date:   Wed May 20 10:42:14 2020 +0100

    Try to handle failures during --test better
```

Appreciate any suggestions to address this.

Thanks!

**Comment by jrtc27** [comment]
QEMU is segfaulting, not cheribuild. What QEMU commit do you have built?

**Comment by nwf** [comment]
This might be https://github.com/CTSRD-CHERI/qemu/issues/102

**Comment by bharadwajy** [comment]
> QEMU is segfaulting, not cheribuild. What QEMU commit do you have built?

Right. QEMU tip:
```
commit 28babb1c682358c3d0da6976b6cb2c411083596b (HEAD -> qemu-cheri, origin/qemu-cheri, origin/HEAD)
Author: Alfredo Mazzinghi <am2419@cl.cam.ac.uk>
Date:   Wed May 20 19:17:11 2020 +0100

    Un-break qemu build.
    
    Fix previous commit to enable instruction tracing for targets that support it.
    Add stubs to other architectures.
    
    This reverts commit fc3d306aee9d49c0f8977dd9bcaa9fdbaae85d34.
```

**Comment by bharadwajy** [comment]
Rerunning cheribuild.py with repo HEAD changed to commit 0f9e2df9e59c68050e4bfc9d667ba27cc35065b9 does not see hit the issue.

**Comment by qwattash** [comment]
I may have broken it with the latest merge, checking right now.

**Comment by qwattash** [comment]
Looks like CTSRD-CHERI/qemu#102 might be very similar to what you are seeing. That seems to be fixed by 28babb1c682358c3d0da6976b6cb2c411083596b, maybe a qemu rebuild with `--clean` helps?

**Comment by bharadwajy** [comment]
> Looks like [CTSRD-CHERI/qemu#102](https://github.com/CTSRD-CHERI/qemu/issues/102) might be very similar to what you are seeing. That seems to be fixed by [CTSRD-CHERI/qemu@28babb1](https://github.com/CTSRD-CHERI/qemu/commit/28babb1c682358c3d0da) maybe a qemu rebuild with `--clean` helps?

Thanks! That seems to have worked.
